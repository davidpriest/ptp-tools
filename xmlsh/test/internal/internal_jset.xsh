# Test of jset

# Simple string

# empty string
jset -v s1 -c java.lang.String
xtype $s1
echo String Value: $s1

# string with value
jset -v s1 -c java.lang.String "Hi"
echo String Value: "(init)" $s1

# Append
jset -v s1 -o $s1 -m concat " There"
echo String Value: "(concat)"  $s1

# Replace (2 args)
jset -v s2 -o $s1 -m replace "Hi" "Bye"
echo String Value: "(replace)" $s2

# Integer values
jset -v i1 -o $s1 -m length
xtype $i1
echo Length: $i1

# Import test
import java ../bin/xmlsh-test.jar

# Default constructor
jset -v t1 -c org.xmlsh.test.TestTypes
xtype $t1
jset -v m1 -o $t1 -m getConstructor

# String constructor
jset -v t1 -c org.xmlsh.test.TestTypes "String"
jset -v m1 -o $t1 -m getConstructor
echo $m1

# Integer constructor
jset -v t1 -c org.xmlsh.test.TestTypes $i1
jset -v m1 -o $t1 -m getConstructor
echo $m1


# Overloaded multi arg constructors
jset -v t2 -c org.xmlsh.test.TestTypes <[ 1 ]> "String"
jset -v m2 -o $t2 -m getConstructor
echo $m2





