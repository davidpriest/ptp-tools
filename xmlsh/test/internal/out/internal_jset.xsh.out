java.lang.String
String Value: 
String Value: (init) Hi
String Value: (concat) Hi There
String Value: (replace) Bye There
java.lang.Integer
Length: 8
org.xmlsh.test.TestTypes
TestTypes(String)
TestTypes(Integer)
TestTypes(Long,String)
